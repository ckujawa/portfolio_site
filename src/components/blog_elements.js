import styled from 'styled-components'

export const Title = styled.h1`
    font-size: 3em;
    line-height: 3.5em;
    text-align: center;
    margin: 1.2rem 0;
`;

export const Paragraph = styled.p`
    font-size: 1em;
    line-height: 1.2em;
    margin: 1rem 0;
    padding: .3rem;
`;

export const Image = styled.img`
    margin: 1.2rem auto;
    padding: .4rem;
    max-width: 90%;
    height: auto;
`

export const ImageDiv = styled.div`
    width: 100%;
    margin: .5rem 0;
    padding: .5rem;
    display: flex;
    justify-content: center;
    align-items: center;
`

export const Code = styled.code`
    display: inline-block;
    padding: 1rem;
    font-family: monospace;
    font-size: 0.9em;
    background-color: #e0d6f9;
    border: 1px solid black;
    border-radius: 5px;
    justify-self: center;
    align-self: center;
`

export const Pre = styled.pre`
    display: grid;
`