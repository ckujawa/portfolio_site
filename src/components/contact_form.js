import React from 'react'
import styled from 'styled-components'

const FormWrapper = styled.div`
    margin-top: 3rem;
    position: relative;
    width: 400px;
    background-color: white;
    margin-left: auto;
    margin-right: auto;
    height: 400px;
    width: 600px;
    border: 1px solid black;
    box-shadow: 3px 4px 20px 3px;
    padding: 0;
`

const FieldContainer = styled.div`
    font-size: 1em;
    position: relative;
    width: 80%;
    margin: 1.7em auto; 

    input{
        width: 100%;
        border: none;
        border-bottom: 1px solid black;
    }

    input:focus{
    outline: none;
    border-bottom: 2px solid #003EAA;
    transition: 0.4s ease all;
    -moz-transition: 0.4s ease all;
    -webkit-transition: 0.4s ease all;
    }

    input:focus + label, textarea:focus + label{
    font-size: .8em;
    color: #595959;
    position: absolute;
    left: 0;
    top: -1.7em;
    transition: 0.4s ease all;
    -moz-transition: 0.4s ease all;
    -webkit-transition: 0.4s ease all;
    }

    input:valid + label, textarea:valid + label{
    font-size: .8em;
    color: #595959;
    position: absolute;
    left: 0;
    top: -1.7em;
    }

    label{
    color: #8c8c8c;
    position: absolute;
    left: 0;
    top: 0;
    }

    textarea{
    width: 100%
    }

    textarea:focus{
    outline: none;
    border: 2px solid #003EAA;  
    transition: 0.4s ease all;
    -moz-transition: 0.4s ease all;
    -webkit-transition: 0.4s ease all;
    }

    
    button {
        font-family: 'Lobster+Two', cursive;
        font-size: 1.3em;
        background-color: #003EAA;
        color: white;
        border: none;
        border-radius: 4px;
        position: absolute;
        right: 2px;
    }
`

const ContactForm = () => (
    <FormWrapper>
        <form name="contact" method="POST" data-netlify="true">
         <input type="hidden" name="form-name" value="contact" />
            <FieldContainer>
                <input type="text" name="name" id="name" required />
                <label for="name">Your Name</label>
            </FieldContainer>
            <FieldContainer>
                <input id="email" name="email" type="email" required />
                <label for="email">Email Address</label>
            </FieldContainer>
            <FieldContainer>
                <input id="phone" name="phone" type="tel" required />
                <label for="phone">Phone Number</label>
            </FieldContainer>
            <FieldContainer>
                <textarea id="request" name="request" rows="4" columns="50" required />
                <label for="request">Your Request</label>
            </FieldContainer>
            <FieldContainer>
                <button type="submit">Submit</button>
            </FieldContainer>
        </form>
    </FormWrapper>
)

export default ContactForm