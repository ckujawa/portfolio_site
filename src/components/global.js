import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
    @import url('https://fonts.googleapis.com/css?family=Arimo|Lato&display=swap');

    * {
        box-sizing: border-box;
        padding: 0;
        margin: 0;
    }   
    
    body{
        margin: 0;
        padding: 0;
        font-family: Lato sans-serif;
        font-size: 1em;
    }

    h1, h2, h3, h4, h5, h6 {
        font-family: Arimo sans-serif;
    }
`;