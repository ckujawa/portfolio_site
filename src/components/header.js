import React from 'react';
import Link from 'gatsby-link';
import styled from 'styled-components';
import { FaTwitter, FaGithub, FaSkype } from 'react-icons/fa';
import Navigator from './navigator';

import logo from '../imgs/logo.svg';

const HeaderWrapper = styled.div`
      background: #0F5263;
      height: 175px;
      margin-bottom: 1.5rem;

      h1{
        margin: 30px 0 0 0;
        img{
          height: 100px;
          margin-bottom: .5rem;
        }
      }
`;

const HeaderContainer = styled.div`
      position: relative;
      margin: 0 auto;
      max-width: 1280px;
      padding: 1.45rem 1.0875rem;
`;

const SocialBlock = styled.div`
    position: absolute;
    top:  10px;
    left: 30px;
    padding-right: 10px;
    font-size: 2rem;
    height: 2rem;
    overflow: hidden;
    margin-right: 5px;
    margin-bottom: 10px;

    a {
        text-decoration: none;
        background-color: #ddb456;
        color: #a07514;
        margin: 0;
        padding: .2rem;
    }

    a:hover{
      background-color: #a07514;
      color: #ddb456;
    }

`;

const Header = ({ siteTitle }) => (
  <HeaderWrapper>
    <HeaderContainer>
      <h1>
        <Link
          to="/"
          style={{
            color: 'white',
            textDecoration: 'none',
          }}
        >
          <img src={logo} alt="Chris Kujawa Web Development logo" />
        </Link>
      </h1>
      <SocialBlock>
        <a href='https://twitter.com/cmkfotoDsgn' title="@cmkfotoDsgn"><FaTwitter /></a>
        <a href='https://github.com/ckujawa' title="ckujawa on Github"><FaGithub /></a>
        <a href='skype:cmkujawa?chat' title='cmkujawa on Skype'><FaSkype /></a>
      </SocialBlock>
      <Navigator isOpen={false}></Navigator>
      {/* <nav>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/blog">Blog</Link>
        </li>
        <li>
          <Link to="/work">Work</Link>
        </li>
      </nav> */}
    </HeaderContainer>
  </HeaderWrapper >

)

export default Header
