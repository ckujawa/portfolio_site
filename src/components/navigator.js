import React from "react";
import Menu from 'react-burger-menu/lib/menus/slide';
import styled from 'styled-components';

const StyledBurgerMenu = styled.div`
    .bm-item {
        text-align:center;  
        display: inline-block;
        text-decoration: none;
        margin-bottom: 5vh;
        color: #d1d1d1;
        transition: color 0.2s;
    }
    .bm-item:hover {
        color: white;
    }
    .bm-burger-button {
        position: fixed;
        width: 45px;
        height: 25px;
        right: 20px;
        top:   10px;
    }
    .bm-burger-bars {
        color: #a07514;
        background: #a07514;;
    }
    .bm-cross-button {
        height: 40px;
        width: 20px;
    }
    .bm-cross {
        background: #bdc3c7;
    }
    .bm-menu {
        position: relative;
        background: rgba(0, 0, 0, 0.3);
        padding: 2.5em 1.5em 0;
        font-size: 2em;
    }
    .bm-morph-shape {
        fill: #373a47;
    }
    .bm-item-list {
        position: static;
        color: #b8b7ad;
    }

    .bm-overlay {
        background: rgba(0, 0, 0, 0.3);
    }
`;

export default () => (
    <StyledBurgerMenu>
        <Menu right>
            <a href="/" className="menu-item" id="home">Home</a>
            <a href="/blog" className="menu-item" id="blog">Blog</a>
            <a href="/work" className="menu-item" id="work">Work</a>
        </Menu>
    </StyledBurgerMenu>
)