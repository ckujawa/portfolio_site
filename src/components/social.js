import React from 'react'
import styled from 'styled-components'

const Social = styled.div`
    padding-right: 10px;
    font-size: 2rem;
    height: 2rem;
    overflow: hidden;
    margin-right: 5px;

    a {
        text-decoration: none;
        background-color: #ddb456;
        color: #a07514;
        margin: 0;
        padding: .2rem;
    }

    a:hover{
      background-color: #a07514;
      color: #ddb456;
    }

`;

export const SocialBlock = ({children}) => {
    return (
        <Social>
            {children}
        </Social>
    )
}