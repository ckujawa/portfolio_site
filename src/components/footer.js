import React from 'react'
import styled from 'styled-components'
import { FaTwitter, FaGithub, FaSkype } from 'react-icons/fa'
import {SocialBlock} from './social'

const Foot = styled.footer`
    height: 50px;
    width: 100%;
    padding: 0;
    margin: 0;
    background: black;
    position: absolute;
    bottom: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const Footer = () => {
    return (<Foot>
        <SocialBlock>
            <a href='https://twitter.com/cmkfotoDsgn' title="@cmkfotoDsgn"><FaTwitter /></a>
            <a href='https://github.com/ckujawa' title="ckujawa on Github"><FaGithub /></a>
            <a href='skype:cmkujawa?chat' title='cmkujawa on Skype'><FaSkype /></a>
        </SocialBlock>
    </Foot>);
}