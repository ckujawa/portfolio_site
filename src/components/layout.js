import React from 'react'
import Helmet from 'react-helmet'
import { StaticQuery, graphql } from 'gatsby'
import Header from '../components/header'
import { Footer } from '../components/footer'
import { GlobalStyle } from '../components/global'
import styled from 'styled-components'

const ChildContainer = styled.div`
  background: #d6defa;
  width: 100%;
  min-height: calc(100vh - 150px);
  padding: 1.5rem;
`

export const PureLayout = ({ children, data }) => {
  console.log({ children })
  console.log({ data })
  return (
    <div>
      <GlobalStyle />
      <Helmet
        title={data.site.siteMetadata.title}
        meta={[
          {
            name: 'description',
            content:
              'The portfolio and blog of Chris Kujawa, a full-stack web developer specializing in creating blazingly fast e-commerce sites with Gatsby and React JS.',
          },
          {
            name: 'keywords',
            content:
              'web development, web developer, ecommerce development, ecommerce website design, ecommerce website development, ecommerce website developer, ecommerce website, custom ecommerce website, GatsbyJS, GatsbyJS Development, GatsbyJS website, Gatsby ecommerce, ReactJS, React developer',
          },
        ]}
      >
        <html lang="en" />
      </Helmet>
      <Header />
      <ChildContainer>{children}</ChildContainer>
      <Footer />
    </div>
  )
}

export const Layout = props => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={data => <PureLayout {...props} data={data} />}
  />
)
