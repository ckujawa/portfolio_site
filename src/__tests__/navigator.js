import React from 'react'
import renderer from 'react-test-renderer'

import { navigator } from '../components/navigator'

describe('navigator', () => {
  it('renders-correctly', () => {
    const tree = renderer.create(<navigator />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
