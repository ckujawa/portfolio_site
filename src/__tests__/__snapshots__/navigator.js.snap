// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigator renders-correctly 1`] = `<navigator />`;
